presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -236,7 +261,59 @@ public boolean hasSingleNodeExecutionPreference(FunctionRegistry functionRegistr
 
     public boolean isStreamable()
     {
-        return !preGroupedSymbols.isEmpty() && groupingSets.size() == 1 && !Iterables.getOnlyElement(groupingSets).isEmpty();
+        return !preGroupedSymbols.isEmpty() && groupingSetCount == 1 && globalGroupingSets.isEmpty();
+    }
+
+    public static GroupingSetDescriptor globalAggregation()
+    {
+        return singleGroupingSet(ImmutableList.of());
+    }
+
+    public static GroupingSetDescriptor singleGroupingSet(List<Symbol> groupingKeys)
+    {
+        Set<Integer> globalGroupingSets;
+        if (groupingKeys.isEmpty()) {
+            globalGroupingSets = ImmutableSet.of(0);
+        }
+        else {
+            globalGroupingSets = ImmutableSet.of();
+        }
+
+        return new GroupingSetDescriptor(groupingKeys, 1, globalGroupingSets);
+    }
+
+    public static GroupingSetDescriptor multipleGroupingSets(List<Symbol> groupingKeys, int groupingSetCount, Set<Integer> globalGroupingSets)
```

Comment:
I see why you would prefer to be explicit about `singleGroupingSet`, but I might call this `groupingSets` because you can consider that to be the 'common' case.